caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditConnection.java

```diff
@@ -35,6 +35,11 @@
 
     public ElasticAuditConnection() throws AuditConfigurationException
     {
+            String elasticProtocol = System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PROTOCOL,
+                    System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PROTOCOL));
+            if (elasticProtocol == null) {
+                elasticProtocol = ElasticAuditConstants.ConfigDefault.CAF_ELASTIC_PROTOCOL;
+            }
```

Comment:
Create a function for this.  Note that the fact that you can't mark `elasticProtocol` as `final` should have been giving you a strong hint that functional decomposition is required.

Reply:
The code to new function